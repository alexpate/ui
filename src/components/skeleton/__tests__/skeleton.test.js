/* @flow */
/* eslint-env jest */
import * as React from 'react';
import {sheet} from 'emotion';
import serializer from 'jest-glamor-react';
import renderer from 'react-test-renderer';

import {
  SkeletonGrid,
  SkeletonList,
  SkeletonText,
  SkeletonButton,
  SkeletonAvatar,
  SkeletonCard,
  SkeletonListItem,
  SkeletonPageHeader,
  SkeletonPageHeaderToolbar,
  SkeletonPageHeaderHeading,
  SkeletonPage,
  SkeletonPaper,
} from '../skeleton';

expect.addSnapshotSerializer(serializer(sheet));

describe('components/skeleton', () => {
  describe('SkeletonAvatar', () => {
    test('should render shallow component ok', () => {
      const element = renderer.create(<SkeletonAvatar size={2} />).toJSON();
      expect(element).toMatchSnapshot();
    });
  });
  describe('SkeletonButton', () => {
    test('should render shallow component ok', () => {
      const element = renderer
        .create(<SkeletonButton square={true} />)
        .toJSON();
      expect(element).toMatchSnapshot();
    });
  });
  describe('SkeletonText', () => {
    test('should render shallow component ok', () => {
      const element = renderer.create(<SkeletonText />).toJSON();
      expect(element).toMatchSnapshot();
    });
  });
  describe('SkeletonList', () => {
    test('should render shallow component ok', () => {
      const element = renderer
        .create(
          <SkeletonList>
            <SkeletonListItem />
            <SkeletonListItem />
            <SkeletonListItem />
          </SkeletonList>
        )
        .toJSON();
      expect(element).toMatchSnapshot();
    });
  });
  describe('SkeletonGrid', () => {
    test('should render shallow component ok', () => {
      const element = renderer
        .create(
          <SkeletonGrid center={false}>
            <SkeletonList center={false}>
              <SkeletonPaper>
                <SkeletonList center={false}>
                  <SkeletonText size={14} heading={true} />
                </SkeletonList>
              </SkeletonPaper>
            </SkeletonList>
            <SkeletonList>
              <SkeletonPaper>
                <SkeletonList center={false}>
                  <SkeletonText heading={true} />
                  <SkeletonText size={14} />
                </SkeletonList>
              </SkeletonPaper>
            </SkeletonList>
          </SkeletonGrid>
        )
        .toJSON();
      expect(element).toMatchSnapshot();
    });
  });
  describe('SkeletonGrid', () => {
    test('should render shallow component ok', () => {
      const element = renderer
        .create(
          <div>
            <SkeletonPageHeader>
              <SkeletonPageHeaderHeading width={1100}>
                <SkeletonText />
                <SkeletonButton square={true} />
              </SkeletonPageHeaderHeading>
              <SkeletonPageHeaderToolbar width={1100}>
                <SkeletonText />
                <SkeletonText />
                <SkeletonText />
              </SkeletonPageHeaderToolbar>
            </SkeletonPageHeader>
            <SkeletonPage width={1100}>
              <SkeletonGrid>
                <SkeletonCard />
                <SkeletonCard />
                <SkeletonCard />
              </SkeletonGrid>
            </SkeletonPage>
          </div>
        )
        .toJSON();
      expect(element).toMatchSnapshot();
    });
  });
});
