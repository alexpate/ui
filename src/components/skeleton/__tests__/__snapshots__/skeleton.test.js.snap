// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/skeleton SkeletonAvatar should render shallow component ok 1`] = `
<div
  className="ui-skeleton-shape ui-skeleton-shape--circle"
  style={
    Object {
      "height": 40,
      "width": 40,
    }
  }
/>
`;

exports[`components/skeleton SkeletonButton should render shallow component ok 1`] = `
<div
  className="ui-skeleton-shape ui-skeleton-shape--square"
  style={
    Object {
      "height": 36,
      "width": 36,
    }
  }
/>
`;

exports[`components/skeleton SkeletonGrid should render shallow component ok 1`] = `
<div
  className="ui-box"
  style={
    Object {
      "alignItems": "stretch",
      "flexDirection": "row",
      "marginLeft": "-25px",
      "marginTop": null,
    }
  }
>
  <div
    className="ui-box"
    style={
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "marginLeft": "25px",
        "marginTop": 0,
      }
    }
  >
    <div
      className="ui-paper ui-paper--level-1 ui-paper--hover-level-1"
      onClick={undefined}
      style={
        Object {
          "flex": undefined,
          "marginLeft": 0,
          "marginTop": "10px",
          "padding": "16px",
        }
      }
    >
      <div
        className="ui-box"
        style={
          Object {
            "alignItems": "stretch",
            "flexDirection": "column",
            "marginLeft": null,
            "marginTop": "-10px",
          }
        }
      >
        <div
          className="ui-skeleton-shape ui-skeleton-shape--square"
          size={14}
          style={
            Object {
              "flex": undefined,
              "height": 24,
              "marginLeft": 0,
              "marginTop": "10px",
              "width": "50%",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="ui-box"
    style={
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "marginLeft": "25px",
        "marginTop": 0,
      }
    }
  >
    <div
      className="ui-paper ui-paper--level-1 ui-paper--hover-level-1"
      onClick={undefined}
      style={
        Object {
          "flex": undefined,
          "marginLeft": 0,
          "marginTop": "10px",
          "padding": "16px",
        }
      }
    >
      <div
        className="ui-box"
        style={
          Object {
            "alignItems": "stretch",
            "flexDirection": "column",
            "marginLeft": null,
            "marginTop": "-10px",
          }
        }
      >
        <div
          className="ui-skeleton-shape ui-skeleton-shape--square"
          style={
            Object {
              "flex": undefined,
              "height": 24,
              "marginLeft": 0,
              "marginTop": "10px",
              "width": "50%",
            }
          }
        />
        <div
          className="ui-skeleton-shape ui-skeleton-shape--square"
          size={14}
          style={
            Object {
              "flex": undefined,
              "height": 12,
              "marginLeft": 0,
              "marginTop": "10px",
              "width": "50%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`components/skeleton SkeletonGrid should render shallow component ok 2`] = `
<div>
  <div
    className="ui-skeleton-page-header"
    style={Object {}}
  >
    <div
      className="ui-skeleton-page-header-heading"
      style={
        Object {
          "width": 1100,
        }
      }
    >
      <div
        className="ui-skeleton-shape ui-skeleton-shape--square"
        style={
          Object {
            "height": 12,
            "width": "50%",
          }
        }
      />
      <div
        className="ui-skeleton-shape ui-skeleton-shape--square"
        style={
          Object {
            "height": 36,
            "width": 36,
          }
        }
      />
    </div>
    <div
      className="ui-skeleton-page-header-toolbar"
      style={Object {}}
    >
      <div
        className="ui-box"
        style={
          Object {
            "margin": "0px auto",
            "width": 1100,
          }
        }
      >
        <div
          className="ui-box"
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 46,
              "marginLeft": "-25px",
              "marginTop": null,
            }
          }
        >
          <div
            className="ui-skeleton-shape ui-skeleton-shape--square"
            style={
              Object {
                "flex": undefined,
                "height": 12,
                "marginLeft": "25px",
                "marginTop": 0,
                "width": "50%",
              }
            }
          />
          <div
            className="ui-skeleton-shape ui-skeleton-shape--square"
            style={
              Object {
                "flex": undefined,
                "height": 12,
                "marginLeft": "25px",
                "marginTop": 0,
                "width": "50%",
              }
            }
          />
          <div
            className="ui-skeleton-shape ui-skeleton-shape--square"
            style={
              Object {
                "flex": undefined,
                "height": 12,
                "marginLeft": "25px",
                "marginTop": 0,
                "width": "50%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui-box"
    style={Object {}}
  >
    <div
      className="ui-box"
      style={
        Object {
          "margin": "50px auto",
          "width": 1100,
        }
      }
    >
      <div
        className="ui-box"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginLeft": "-25px",
            "marginTop": null,
          }
        }
      >
        <div
          className="ui-paper ui-paper--level-1 ui-paper--hover-level-1"
          onClick={undefined}
          style={
            Object {
              "flex": 1,
              "marginLeft": "25px",
              "marginTop": 0,
              "padding": "16px",
              "paddingBottom": "150px",
              "paddingTop": "75px",
            }
          }
        >
          <div
            className="ui-box"
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "marginLeft": null,
                "marginTop": "-25px",
              }
            }
          >
            <div
              className="ui-skeleton-shape ui-skeleton-shape--circle"
              style={
                Object {
                  "flex": undefined,
                  "height": 80,
                  "marginLeft": 0,
                  "marginTop": "25px",
                  "width": 80,
                }
              }
            />
            <div
              className="ui-skeleton-shape ui-skeleton-shape--square"
              style={
                Object {
                  "flex": undefined,
                  "height": 12,
                  "marginLeft": 0,
                  "marginTop": "25px",
                  "width": "14%",
                }
              }
            />
            <div
              className="ui-skeleton-shape ui-skeleton-shape--square"
              style={
                Object {
                  "flex": undefined,
                  "height": 12,
                  "marginLeft": 0,
                  "marginTop": "25px",
                  "width": "20%",
                }
              }
            />
          </div>
        </div>
        <div
          className="ui-paper ui-paper--level-1 ui-paper--hover-level-1"
          onClick={undefined}
          style={
            Object {
              "flex": 1,
              "marginLeft": "25px",
              "marginTop": 0,
              "padding": "16px",
              "paddingBottom": "150px",
              "paddingTop": "75px",
            }
          }
        >
          <div
            className="ui-box"
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "marginLeft": null,
                "marginTop": "-25px",
              }
            }
          >
            <div
              className="ui-skeleton-shape ui-skeleton-shape--circle"
              style={
                Object {
                  "flex": undefined,
                  "height": 80,
                  "marginLeft": 0,
                  "marginTop": "25px",
                  "width": 80,
                }
              }
            />
            <div
              className="ui-skeleton-shape ui-skeleton-shape--square"
              style={
                Object {
                  "flex": undefined,
                  "height": 12,
                  "marginLeft": 0,
                  "marginTop": "25px",
                  "width": "14%",
                }
              }
            />
            <div
              className="ui-skeleton-shape ui-skeleton-shape--square"
              style={
                Object {
                  "flex": undefined,
                  "height": 12,
                  "marginLeft": 0,
                  "marginTop": "25px",
                  "width": "20%",
                }
              }
            />
          </div>
        </div>
        <div
          className="ui-paper ui-paper--level-1 ui-paper--hover-level-1"
          onClick={undefined}
          style={
            Object {
              "flex": 1,
              "marginLeft": "25px",
              "marginTop": 0,
              "padding": "16px",
              "paddingBottom": "150px",
              "paddingTop": "75px",
            }
          }
        >
          <div
            className="ui-box"
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "marginLeft": null,
                "marginTop": "-25px",
              }
            }
          >
            <div
              className="ui-skeleton-shape ui-skeleton-shape--circle"
              style={
                Object {
                  "flex": undefined,
                  "height": 80,
                  "marginLeft": 0,
                  "marginTop": "25px",
                  "width": 80,
                }
              }
            />
            <div
              className="ui-skeleton-shape ui-skeleton-shape--square"
              style={
                Object {
                  "flex": undefined,
                  "height": 12,
                  "marginLeft": 0,
                  "marginTop": "25px",
                  "width": "14%",
                }
              }
            />
            <div
              className="ui-skeleton-shape ui-skeleton-shape--square"
              style={
                Object {
                  "flex": undefined,
                  "height": 12,
                  "marginLeft": 0,
                  "marginTop": "25px",
                  "width": "20%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/skeleton SkeletonList should render shallow component ok 1`] = `
<div
  className="ui-box"
  style={
    Object {
      "alignItems": "stretch",
      "flexDirection": "column",
      "marginLeft": null,
      "marginTop": "-10px",
    }
  }
>
  <div
    className="ui-paper ui-paper--level-1 ui-paper--hover-level-1"
    onClick={undefined}
    style={
      Object {
        "flex": undefined,
        "marginLeft": 0,
        "marginTop": "10px",
        "padding": "25px",
      }
    }
  >
    <div
      className="ui-box"
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": "-25px",
          "marginTop": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        className="ui-skeleton-shape ui-skeleton-shape--square"
        style={
          Object {
            "flex": undefined,
            "height": 12,
            "marginLeft": "25px",
            "marginTop": 0,
            "width": "20%",
          }
        }
      />
      <div
        className="ui-skeleton-shape ui-skeleton-shape--square"
        style={
          Object {
            "flex": undefined,
            "height": 12,
            "marginLeft": "25px",
            "marginTop": 0,
            "width": "10%",
          }
        }
      />
    </div>
  </div>
  <div
    className="ui-paper ui-paper--level-1 ui-paper--hover-level-1"
    onClick={undefined}
    style={
      Object {
        "flex": undefined,
        "marginLeft": 0,
        "marginTop": "10px",
        "padding": "25px",
      }
    }
  >
    <div
      className="ui-box"
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": "-25px",
          "marginTop": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        className="ui-skeleton-shape ui-skeleton-shape--square"
        style={
          Object {
            "flex": undefined,
            "height": 12,
            "marginLeft": "25px",
            "marginTop": 0,
            "width": "20%",
          }
        }
      />
      <div
        className="ui-skeleton-shape ui-skeleton-shape--square"
        style={
          Object {
            "flex": undefined,
            "height": 12,
            "marginLeft": "25px",
            "marginTop": 0,
            "width": "10%",
          }
        }
      />
    </div>
  </div>
  <div
    className="ui-paper ui-paper--level-1 ui-paper--hover-level-1"
    onClick={undefined}
    style={
      Object {
        "flex": undefined,
        "marginLeft": 0,
        "marginTop": "10px",
        "padding": "25px",
      }
    }
  >
    <div
      className="ui-box"
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": "-25px",
          "marginTop": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        className="ui-skeleton-shape ui-skeleton-shape--square"
        style={
          Object {
            "flex": undefined,
            "height": 12,
            "marginLeft": "25px",
            "marginTop": 0,
            "width": "20%",
          }
        }
      />
      <div
        className="ui-skeleton-shape ui-skeleton-shape--square"
        style={
          Object {
            "flex": undefined,
            "height": 12,
            "marginLeft": "25px",
            "marginTop": 0,
            "width": "10%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`components/skeleton SkeletonText should render shallow component ok 1`] = `
<div
  className="ui-skeleton-shape ui-skeleton-shape--square"
  style={
    Object {
      "height": 12,
      "width": "50%",
    }
  }
/>
`;
